LDAR_Sim/model_code/generic_functions.py

```diff
@@ -283,3 +283,44 @@ def quick_cal_daylight(date,lat,lon):
     sunset = ss
 
     return (sunrise,sunset)
+
+
+def find_homebase(x1,y1,HX,HY): 
+    XY = list(zip(HX,HY))
+    D = []
+    for xy in XY:
+        x2 = xy[0]
+        y2 = xy[1]
+        d = ((x1 - x2)**2 + (y1 - y2)**2)**0.5
+        D.append(d)
+    ind =D.index(min(D))
+    dist = min(D) * 110
+    return (XY[ind],dist)
+
+def get_distance (x1,y1,x2,y2,form): 
+    if form == "Euclidian": 
+        d = ((x1 - x2)**2 + (y1-y2)**2)**0.5
+        d = d * 110 # covert to km  
+    elif form == "route":
+        d = 100 
+    return d 
```

Comment:
Why does this return 100 always in routing 'form'?

Reply:
@AkaMrmuffin

Reply:
Thanks for pointing this out. The "elif" is just a placeholder, which will be replaced with a function that calculates the distance based on the road network in the future. For now, both aircraft and truck&OGI are using the Euclidian distance. I absolutely agree that haversine distance is better. I will upgrade this get_distance function in the future. I don't think it matters now since this function is only for find the nearest distance. But, Yes, lets switch to Haversine distance.